src/nccl_ofi_net.c

```diff
@@ -1310,14 +1310,64 @@ static ncclResult_t ofi_listen(int dev, void *handle, void **listenComm)
 	return ret;
 }
 
+struct connection_info {
+	char ep_name[MAX_EP_ADDR];
+	size_t ep_namelen;
+	nccl_ofi_req_t* req;
+	struct connection_info* memory_buffer;
+	bool connect_to_self;
+};
+
+struct connection_info* allocate_connection_info(int dev, nccl_ofi_req_t *req, char* remote_ep_name)
+{
+	struct connection_info* conn_info = (struct connection_info*)calloc(1, sizeof(*conn_info));
+	if (OFI_UNLIKELY(conn_info == NULL)) {
+		return NULL;
+	}
+	conn_info->ep_namelen = sizeof(conn_info->ep_name);
+
+	/* Get local EP address to transfer in the connect message */
+	int ret = fi_getname(&(nccl_ofi_component[dev]->ep->fid),
+			 (void *)conn_info->ep_name,
+			 &(conn_info->ep_namelen));
+	if (ret != 0) {
+		NCCL_OFI_WARN("Call to fi_getname() failed with RC: %d, ERROR: %s",
+			      ret, fi_strerror(-ret));
+		free(conn_info);
+		return NULL;
+	}
+
+	conn_info->req = req;
+	/* for future freeing conn_info buffer */
+	conn_info->memory_buffer = conn_info;
+	conn_info->connect_to_self = (memcmp(conn_info->ep_name, remote_ep_name, conn_info->ep_namelen) == 0);
+
+	return conn_info;
+}
+
+ncclResult_t finalize_connection_info(struct connection_info* conn_info)
+{
+	/* process postponed completions */
+	do {
+		ncclResult_t ret = nccl_ofi_progress(nccl_ofi_component[conn_info->req->dev]);
+		if (OFI_UNLIKELY(ret != 0)) {
+			return ret;
+		}
+	} while (conn_info->req->state != NCCL_OFI_REQ_COMPLETED);
+	free_nccl_ofi_req(conn_info->req, false);
+	free(conn_info->memory_buffer);
+	NCCL_OFI_TRACE(NCCL_NET, "Completions processed");
```

Comment:
Prefer a better log message like "Processed completion for connect message"

Reply:
done